subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt.c

```diff
@@ -278,13 +278,10 @@ img_mgmt_erase(struct mgmt_ctxt *ctxt)
 		}
 	}
 
-	rc = img_mgmt_impl_erase_slot(slot);
-
-	if (rc != 0) {
+	if (img_mgmt_impl_erase_slot(slot) != 0) {
```

Comment:
This should return the result code directly, e.g. for the code I'm working on this function can return busy